lib/datadog/core/transport/http/adapters/net.rb

```diff
@@ -86,6 +86,21 @@ def url
               "http://#{hostname}:#{port}?timeout=#{timeout}"
             end
 
+            def net_http_path_from_env(env)
+              path = env.path
+              case query = env.query
+              when Hash
+                path = path + '?' + URI.encode_www_form(query)
+              when String
+                path = path + '?' + query
+              when nil
+                # Nothing
+              else
+                raise ArgumentError, "Invalid type for query: #{query}"
+              end
+              path
```

Comment:
You can avoid that by just returning path in `nil` case and skip assignment in other cases

```suggestion
              case query = env.query
              when Hash then path + '?' + URI.encode_www_form(query)
              when String then path + '?' + query
              when nil then path
              else raise ArgumentError, "Invalid type for query: #{query}"
              end
```